### ISSUE TITLE
Request: Internet Connection Detection

### ISSUE THREAD

**Issue by TotallyNotInUse** [ytmdesktop/ytmdesktop#215]
Hello, I have set the app to open with Windows but sometimes I don't connect to the internet so when I re-connect and open the app it shows that I need internet for YTM to work and I need to close completely the app and reopen so it refreshes.

I think a button to refresh or a key to refresh should be added. It also could auto-refresh when a new connection is detected.

**Comment by TotallyNotInUse** [comment]
May I ask if this is being developed right now?

**Comment by sli** [comment]
I run into this regularly because YTMDP will load up before my VPN has connected. Even just a simple retry button would suffice.

**Comment by AngryJKirk** [comment]
Yeah, I suffer from the same issue, just a refresh button would be enough to solve this problem.

**Comment by sur0g** [comment]
This is the main issue for me because I use the internet tethering from the phone which I regularly disconnect from the computer.

**Comment by Mizumaky** [comment]
Same here.

**Comment by Mizumaky** [comment]
#449 another copy of this issue, already marked as a bug

**Comment by TotallyNotInUse** [comment]
> #449 another copy of this issue, already marked as a bug
> (skip to the "---><---" in the desc, text before it is just a copy of another badly written issue)

Well this is issue is pretty old (older than the one you linked) so it's not weird to see multiple copies of it.

**Comment by jippylong12** [comment]
Tested this on the latest development branch and wasn't able to reproduce the original issue. YouTube Music's own web app now handles the offline case with a "Connect to the internet" page that includes a Retry button, a "Go to downloads" option, and auto-reconnects when connectivity returns.

If the YTMD loading screen itself ever gets stuck on a load error (before YTM's page loads), a retry button could be added to `YTMViewLoading.vue` that calls the existing `ytmViewNavigateDefault` IPC handler, along with resetting the error state in the main process. But as of now, YTM's own offline handling seems to cover it.

This might be a good candidate to close since the underlying issue appears to be resolved on YouTube Music's side.
